### PR TITLE
Added Unit tests for branching/merging of IO ports

### PR DESCRIPTION
**Pull request by PhilippPlank**
Unit tests which check merging and branching of InPorts and OutPorts in ProcessModels and hierarchical ProcessModels.

### CODE REVIEW

File: tests/lava/magma/runtime/test_io_ports.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.var = Var(shape=(2,), init=5)
+        self.inp = InPort(shape=(2,))
+
+
+# A minimal hierarchical PyProcModel implementing HP2
+@implements(proc=HP1, protocol=LoihiProtocol)
+class PyProcModelHP1(AbstractSubProcessModel):
+    h_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, int)
```
Comment (awintel) [inline tests/lava/magma/runtime/test_io_ports.py:55]: Hierarchical processes do not have LavaPyTypes. Delete this.

File: tests/lava/magma/runtime/test_io_ports.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+# A minimal process with a Var and an InPort
+class P2(AbstractProcess):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.var = Var(shape=(2,), init=5)
+        self.inp = InPort(shape=(2,))
+
+
+# A minimal hierarchical PyProcModel implementing HP2
+@implements(proc=HP1, protocol=LoihiProtocol)
```
Comment (awintel) [inline tests/lava/magma/runtime/test_io_ports.py:53]: No protocol required. Will be derived from child processes.

File: tests/lava/magma/runtime/test_io_ports.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+        # Connect the OutPort of the hierarchical process with the OutPort of
+        # the nested process
+        self.p1 = P1()
+        self.p1.out.connect(proc.out_ports.h_out)
+
+
+# A minimal hierarchical PyProcModel implementing HP2
+@implements(proc=HP2, protocol=LoihiProtocol)
+class PyProcModelHP2(AbstractSubProcessModel):
+    h_var: np.ndarray = LavaPyType(np.ndarray, np.int32)
```
Comment (awintel) [inline tests/lava/magma/runtime/test_io_ports.py:69]: Same here.

File: tests/lava/magma/runtime/test_io_ports.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+        """Builds sub Process structure of the Process."""
+
+        # Connect the OutPort of the hierarchical process with the OutPort of
+        # the nested process
+        self.p1 = P1()
+        self.p1.out.connect(proc.out_ports.h_out)
+
+
+# A minimal hierarchical PyProcModel implementing HP2
+@implements(proc=HP2, protocol=LoihiProtocol)
```
Comment (awintel) [inline tests/lava/magma/runtime/test_io_ports.py:67]: Same here.

**Review by awintel — APPROVED** [review]
Thanks for adding unit tests for these features. 

My comments are not necessarily gating for this release. It's just about removing unnecessary code for hierarchical processes. If you want you can also do it later.

**Review by mathisrichter — APPROVED** [review]
Looks good

**Pull request merged**